app/controllers/import_controller.rb

```diff
@@ -297,6 +298,15 @@ def import_phone_field(value)
     e_164.blank? ? value : e_164
   end
 
+  def import_and_validate_int_phone_field(field, value, row_ind)
+    unless value == value&.gsub(/[^0-9.\-()+ ]/, '')
+      raise ValidationError.new("Value '#{value}' for '#{VALIDATION[field][:label]}' is not an acceptable value, please only use valid characters which \
+                                include digits, \".\", \"-\", \"(\", \")\", \"+\", \" \".", row_ind)
```

Comment:
This validation needs to happen from the API as well. It's kinda tough, because it would be nice if we could do it on the Patient model, but then it is difficult to figure out when we would convert the number to e164 format.

But this raises a bigger question for me, are we sure we need to log a validation error when we find other characters that we want to ignore? For a primary or secondary phone, if someone puts something like "hello" in the middle of a phone number, the `Phonelib` library just ignores the "hello". What is a little confusing in this case is that we say that something like "-" is allowed, but the `Phonelib` library is going to filter those characters out anyway when it parses it. However, in the UI, you can add a "-" character and have it persist. I feel like ideally all of this would be consistent. I would think it would make the most sense to either:
* Never use `Phonelib`, just validate that we only have the characters we want, and then preserve the string exactly as entered if it is valid (how the UI works right now for international telephone)
* Only use `Phonelib`, and let `Phonelib` just silently filter out garbage characters, and don't worry about logging a validation error (similar to how primary/secondary phone work on import now)

What do you think?

Reply:
I agree with these concerns and think both of these are viable options, would probably leave it up to the PO team to see how they would want it to work

Reply:
Yeah I think it makes sense to ask the PO team what their preference is on this.